tests/clients/test_experiment.py

```diff
@@ -363,3 +424,80 @@ def test_experiment_client_list_runs_page(mocker):
         schema_mock.return_value,
         params=[("start", 20), ("limit", 10)],
     )
+
+
+def test_log_run_data(mocker):
+    mocker.patch.object(ExperimentClient, "_patch_raw")
+    run_data_schema_mock = mocker.patch(
+        "faculty.clients.experiment.ExperimentRunDataSchema"
+    )
+    run_data_dump_mock = run_data_schema_mock.return_value.dump
+
+    client = ExperimentClient(mocker.Mock())
+    client.log_run_data(
+        PROJECT_ID,
+        EXPERIMENT_RUN_ID,
+        metrics=[METRIC],
+        params=[PARAM],
+        tags=[TAG],
+    )
+    expected_body = {"metrics": [METRIC], "params": [PARAM], "tags": [TAG]}
+
+    run_data_schema_mock.assert_called_once_with()
+    run_data_dump_mock.assert_called_once_with(expected_body)
+    ExperimentClient._patch_raw.assert_called_once_with(
+        "/project/{}/run/{}/data".format(PROJECT_ID, EXPERIMENT_RUN_ID),
+        json=run_data_dump_mock.return_value,
+    )
+
+
+def test_log_run_data_param_conflict(mocker):
+    expected_message = "bad params"
+    expected_error_code = "conflicting_params"
+    json_mock = mocker.Mock()
+    json_mock.json.return_value = {"parameterKeys": ["bad-key"]}
+    response_mock = Conflict(json_mock, expected_message, expected_error_code)
+
+    mocker.patch.object(
+        ExperimentClient, "_patch_raw", side_effect=response_mock
+    )
+    run_data_schema_mock = mocker.patch(
+        "faculty.clients.experiment.ExperimentRunDataSchema"
+    )
+
+    client = ExperimentClient(mocker.Mock())
+
+    with pytest.raises(ParamConflict, match=expected_message):
+        client.log_run_data(PROJECT_ID, EXPERIMENT_RUN_ID, params=[PARAM])
+    run_data_schema_mock.assert_called_once_with()
```

Comment:
Could probably remove this assertion since we're not checking if the schema's `dump` method call either. This would mean we don't need to bind the `run_data_schema_mock` above either.

Reply:
👍

Reply:
You marked this as resolved but the change does not appear to have been applied to this test (though it has been applied to the later ones)

Reply:
Done now